.github/workflows/benchmark.yml

```diff
@@ -12,35 +18,83 @@ jobs:
   
     if: | 
       startsWith(github.event.comment.body, '/benchmark') && 
-      github.event.issue.pull_request 
+      github.event.issue.pull_request || 
+      github.event_name == 'workflow_dispatch'
 
     steps:
       - name: Checkout
         uses: actions/checkout@v3.3.0
 
+
+      - name: Detect GitHub Event 
+        id: detect-event
+        uses: actions/github-script@v6
+        with:
+          script: |
+            function hasBenchmarkComment() {
+              return 
+                startsWith(github.event.comment.body, '/benchmark') && 
+                github.event.issue.pull_request;
+            }
+
+            function workflowWasDispatched() {
+              return context.eventName == "workflow_dispatch"
+            }
+
+            if (hasBenchmarkComment()) {
+              core.setOutput("event", "issue_comment");
+            } else if (workflowWasDispatched()) {
+              core.setOutput("event", "workflow_dispatch");
+            } else {
+              core.setFailed("An unexpeced event triggered this workflow.");
+            }
+
+
       - name: React with Rocket 
         uses: actions/github-script@v6
+        if: steps.detect-event.outputs.is_comment
         with:
           script: |
-            github.rest.reactions.createForIssueComment({
-              owner: context.issue.owner,
-              repo: context.issue.repo,
-              comment_id: context.payload.comment.id,
-              content: "rocket"
-            });  
+            async function reactWithRocket() {
+              github.rest.reactions.createForIssueComment({
+                owner: context.issue.owner,
+                repo: context.issue.repo,
+                comment_id: context.payload.comment.id,
+                content: "rocket"
+              });  
+            }
+
+            if (core.getInput("event") === "issue_comment") {
+              await reactWithRocket();
+            } else {
+              core.debug("Workflow dispacted from GitHub GUI, not reacting with Rocket.")
```

Comment:
This should be impossible given the `if` guard in the step, right? So maybe it should fail?

Reply:
Good catch, I'll remove the `if:`